integration_tests/README.md

```diff
@@ -5,41 +5,161 @@ are intended to be able to be run against any Spark-compatible cluster/release t
 verify that the plugin is doing the right thing in as many cases as possible.
 
 There are two sets of tests. The PySpark tests are described on this page. The scala tests are
-described [here](../tests/README.md). 
+described [here](../tests/README.md).
 
-## Dependencies
+## Setting Environment
 
 The tests are based off of `pyspark` and `pytest` running on Python 3. There really are
 only a small number of Python dependencies that you need to install for the tests. The
 dependencies also only need to be on the driver.  You can install them on all nodes
 in the cluster but it is not required.
 
-### pytest
-`pip install pytest`
+### Prerequisities
+
+The build requires `OpenJDK 8`, `maven`, and `python`.
+skip to the next section if you have already installed them.
+
+<details><summary><b>Java Environment</b></summary>
+<p>
+
+It is recommended to use `alternatives` to manage multiple java versions.
+Then you can simply set `JAVA_HOME` to JDK directory:
+
+  ```shell script
+  JAVA_HOME=$(readlink -nf $(which java) | xargs dirname | xargs dirname | xargs dirname)
+  ```
+
+</p>
+</details>
 
-Should be enough to get the basics started.
-
-### sre_yield
-`pip install sre_yield`
-
-`sre_yield` provides a set of APIs to generate string data from a regular expression.
-
-### pandas
-`pip install pandas`
-
-`pandas` is a fast, powerful, flexible and easy to use open source data analysis and manipulation tool and is
-only needed when testing integration with pandas.
-
-### pyarrow
-`pip install pyarrow`
-
-`pyarrow` provides a Python API for functionality provided by the Arrow C++ libraries, along with tools for Arrow
-integration and interoperability with pandas, NumPy, and other software in the Python ecosystem. This is used
-to test improved transfer performance to pandas based user defined functions.
-
-## pytest-xdist and findspark
-
-`pytest-xdist` and `findspark` can be used to speed up running the tests by running them in parallel.
+<details><summary><b>Installing python using pyenv</b></summary>
+<p>
```

Comment:
Curious, do we need the `<b>` and `<p>` tags?  For the former we would normally use `*` to mark bold text, and the latter is normally done with simple newlines.  One of the advantages of Markdown is it doesn't have a lot of HTML tag pollution in the text, and it would be nice to minimize it here as well for those reading the README in a terminal window rather than rendered on Github.  I see we're using Markdown inside here, so I'm wondering if some of these HTML tags are strictly necessary.

Reply:
The html tags would make the header collapsible.
Previously, `<p>` had to be there to be parse the feature.
I removed the collapsible headers to avoid breaking the coding style.